litex/soc/software/bios/boot.h

```diff
@@ -12,5 +12,8 @@ void flashboot(void);
 void romboot(void);
 void sdcardboot(void);
 void sataboot(void);
+extern void target_init(void) __attribute__((weak));
+extern void target_boot(void) __attribute__((weak));
```

Comment:
why not something like:
```
void target_nop() {}
void target_init(void) __attribute__((weak, alias("target_nop")));
void target_boot(void) __attribute__((weak, alias("target_nop")));
```
to avoid having to test if function exist

Reply:
Yes that's another alternative I onsidered.I like the fact that the test makes it's clear that it's optional, but that's personal preference.
There doesn't appear to have been any goal to minimise code size in the bios so i opted for clarity.